tests/e2e/tests/test_transport_v3.go

```diff
@@ -466,16 +477,96 @@ func testTransportV3SearchServiceTravelWithoutArrivalDate(
 	require.NotEmpty(t, resp.Results[0].TravellingTrips[0].Segments[0].Info.Departure, "unexpected empty response Results[0].TravellingTrips[0].Segments[0].Info.Departure")
 
 	require.True(t, proto.Equal(departureLocationCode, resp.Results[0].TravellingTrips[0].Segments[0].Info.Departure.Location.GetLocationCode()), "unexpected departure location code")
-
-	require.NoError(t, err)
-
 	// Now extract all the values needed for the validate step which comes next
 	require.NotEmpty(t, resp.Metadata, "unexpected empty response Metadata")
 	require.NotEmpty(t, resp.Metadata.SearchId, "unexpected empty response Metadata.SearchId")
 	require.NotEmpty(t, resp.Metadata.SearchId.Value, "unexpected empty response Metadata.SearchId.Value")
 	require.NotEmpty(t, resp.Results[0].ResultId, "unexpected empty response Results[1].ResultId")
 }
 
+func testTransportV3SearchServiceWithoutArrivalLocation(
+	ctx context.Context,
+	t *testing.T,
+	tt *Test,
+	distributorBot *bot.Bot,
+	supplierBot *bot.Bot,
+	productListResponse *transportv3.TransportProductListResponse,
+) {
+	// Extract the filters from the product list response which double also
+	// as the expected results later
+	// The product list request has already made sure that there are 2 results
+	// And that the 2nd result has 2 segments. So just extract the values here
+	firstSegmentDeparture := productListResponse.Trips[2].Segments[0].Departure
+	lastSegmentArrival := productListResponse.Trips[2].Segments[1].Arrival
+
+	departureDate := time.Unix(firstSegmentDeparture.DateTime.Seconds, 0)
+	departureLocationCode := firstSegmentDeparture.Location.GetLocationCode()
+	arrivalDate := time.Unix(lastSegmentArrival.DateTime.Seconds, 0)
+
+	req := &transportv3.TransportSearchRequest{
+		Header: &typesv1.RequestHeader{BaseHeader: &typesv1.Header{}},
+		SearchParameters: &typesv3.SearchParameters{
+			Currency: &typesv3.Currency{
+				Currency: &typesv3.Currency_IsoCurrency{
+					IsoCurrency: typesv3.IsoCurrency(*typesv2.IsoCurrency_ISO_CURRENCY_EUR.Enum()),
```

Comment:
A typesV3 Iso Currency using the typesV2 enum value of the Iso Currency - are you sure that this is correct?